### PR TITLE
Rhmap 11672

### PR DESCRIPTION
**Pull request by pb82**
# Motivation

Extends the fh-db API with a MongoDB compatible layer on top of shared and dedicated databases. This API can be consumed by the new backend service for the databrowser and provides easy replaceability with the actual MongoDB driver as soon as we don't have to support shared dbs.

The API supports the exact MongoDB driver semantics, including promises and cursors. Naming conventions for shared dbs are adhered in all API calls. The following functions are implemented:

```
#createCollection
#dropCollection
#listCollections
#collection
#createIndex
#renameCollection
#close
``` 

Advanced API for CRUD operations can be easily used for shared and dedicated dbs by using the `collection` function and running queries on collections:

```javascript
var localdb = require('fh-db');
localdb.createMongoCompatApi({
  __dbperapp: <true|false>,
  __fhdb: <passed by fh-mbaas-api, only required for shared db>,
  connectionUrl: <connection url for shared or dedicated database>
}).then(function (db) {
  // db is now either a native mongodb connection or a compatible wrapper API
  db.collection("user").find(...);
});

```

ping @wei-lee @rachael-oregan @sedroche

### CODE REVIEW

File: lib/fhmongodb.js [hunk 1]
```diff
@@ -477,4 +496,54 @@ Database.prototype.collection = function (name, cb) {
...
+  if (!this.db) {
+    throw new Error("no database open");
+  }
+
+  var cursor = this.db.listCollections(filter, options);
+  var databaseName = this.db.databaseName;
+
+  // MongoDB `CommandCursor` like object
+  function CommandCursor(cursor) {
+    this.toArray = function (callback) {
```
Comment (fh-qube) [inline lib/fhmongodb.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/ditcher.js [hunk 2]
```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
...
+    console.log(params);
+    console.log(appname_regex.test(params.__fhdb));
+
+    throw new Error("Incorrect parameters for listing collections");
+  }
+
+  var cursor = self.database.listCollections(filter, params);
+
+  // MongoDB `CommandCursor` like object
+  function CommandCursor(cursor) {
```
Comment (fh-qube) [inline lib/ditcher.js]: INFO Rename this 'CommandCursor' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS100)

File: lib/ditcher.js [hunk 3]
```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
...
+    this.toArray = function (callback) {
```
Comment (fh-qube) [inline lib/ditcher.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/ditcher.js [hunk 4]
```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
...
+// adheres to the shared-db naming conventions. Also accepts a filter object to
+// filter the returned collections.
+Ditcher.prototype.doGetCollectionsListCursor = function (filter, params) {
+  var self = this;
+
+  if (!params.__fhdb ||
+    (!params.__dbperapp && !appname_regex.test(params.__fhdb)) ||
+    (params.__dbperapp ===true && params.__fhdb !== self.database.name)) {
+    console.log(params);
+    console.log(appname_regex.test(params.__fhdb));
```
Comment (fh-qube) [inline lib/ditcher.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: lib/ditcher.js [hunk 5]
```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
...
+  return this.database.getCollectionInstance(coll, params, callback);
+};
 
+// Returns a cursor that can be used for listing the collections. This function
+// adheres to the shared-db naming conventions. Also accepts a filter object to
+// filter the returned collections.
+Ditcher.prototype.doGetCollectionsListCursor = function (filter, params) {
+  var self = this;
+
+  if (!params.__fhdb ||
```
Comment (fh-qube) [inline lib/ditcher.js]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS1067)

File: lib/ditcher.js [hunk 6]
```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
...
+    (!params.__dbperapp && !appname_regex.test(params.__fhdb)) ||
+    (params.__dbperapp ===true && params.__fhdb !== self.database.name)) {
+    console.log(params);
```
Comment (fh-qube) [inline lib/ditcher.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: lib/mongo_compat_api.js [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+
+    /**
+     * List all collections
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#listCollections
+     * @param filter Query to filter collections by
+     * @param options (optional) Optional settings
+     * @returns Cursor
+     */
+    this.listCollections = function (filter, options) {
+      var options = this.copyDefaultOptions(options || {});
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MAJOR Rename "options" as this name is already used in declaration at line 100. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)

File: lib/mongo_compat_api.js [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#collection
+     * @param name Query to filter collections by
+     * @param options (optional) Optional settings
+     * @param callback (optional) result callback
+     * @returns Collection instance if no callback passed
+     */
+    this.collection = function (name, options, callback) {
+      assert.ok(name, "collection name required");
+
+      var options = this.copyDefaultOptions(options || {});
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MAJOR Rename "options" as this name is already used in declaration at line 113. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)

File: lib/mongo_compat_api.js [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+    this.createIndex = function (name, fieldOrSpec, options, callback) {
+      assert.ok(name, "collection name required");
+      assert.ok(fieldOrSpec, "index spec required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      var options = this.copyDefaultOptions(options || {});
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MAJOR Rename "options" as this name is already used in declaration at line 131. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)

File: lib/mongo_compat_api.js [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+    this.renameCollection = function (fromCollection, toCollection, options, callback) {
+      assert.ok(fromCollection, "current collection name required");
+      assert.ok(toCollection, "new collection name required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      var options = this.copyDefaultOptions(options || {});
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MAJOR Rename "options" as this name is already used in declaration at line 167. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)

File: lib/mongo_compat_api.js [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+
+/**
+ * Implements a MongoDB Node.js 2.0 driver compatible API that works for
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (ditcher, params) {
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MINOR This function has 311 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
MINOR Function has a complexity of 63 which is greater than 20 authorized. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 14 issues:
* BLOCKER 2 blocker
* MAJOR 8 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/d0b048608341a501f95234ca7a96348388a1efa5/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 6 issues:
* BLOCKER 1 blocker
* MAJOR 1 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/88afbb2787d15617fa8f3fe704b8b9f7e0ebcdbb/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 1 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/f0086554342595c2e6189523a220d24e94a2fbe9/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

File: lib/mongo_compat_api.js [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+            resolve(result);
+          });
+        });
+      }
+    };
+  } else {
+    // The MongoDB compatible API for per-app databases. Ideally it would be enough to just
+    // return the `#ditcher.database.db` object here (the mongo handle). However this results
+    // in a `promiseLibrary not defined` error (with or without bluebird). So we are wrapping
+    // the fhmongodb calls here.
```
Comment (wei-lee) [inline lib/mongo_compat_api.js]: I think it worth spending sometime to investigate the issue. Ideally we don't want to wrapper the mongodb module for apps that have standalone dbs.

File: lib/mongo_compat_api.js (hunk 11 above)
Comment (wei-lee) [inline lib/mongo_compat_api.js]: do we really need to pass `ditcher` here? would it better if we just pass the db configuration here? This module itself can then create the `ditcher` instance for shared app dbs, but then use mongodb client directly for apps with standalone dbs?

File: lib/utils.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+  result.database = {};
+  result.database.driver_options = {};
+  result.database.auth = {};
+  result.database.auth.user = authUser;
+  result.database.auth.pass = authPassword;
+  result.database.auth.source = databaseName;
+  result.database.name = databaseName;
+
+  assert.ok(authUser != undefined);
```
Comment (fh-qube) [inline lib/utils.js]: BLOCKER Replace "!=" with "!==". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: lib/utils.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+  assert.ok(authPassword != undefined);
```
Comment (fh-qube) [inline lib/utils.js]: BLOCKER Replace "!=" with "!==". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: lib/utils.js [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+  assert.ok(databaseName != undefined);
```
Comment (fh-qube) [inline lib/utils.js]: BLOCKER Replace "!=" with "!==". [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEqEqEq)

File: lib/mongo_compat_api.js [hunk 16]
```diff
@@ -0,0 +1,280 @@
...
+
+/**
+ * Implements a MongoDB Node.js 2.0 driver compatible API that works for
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (params) {
```
Comment (fh-qube) [inline lib/mongo_compat_api.js]: MINOR This function has 194 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
MINOR Function has a complexity of 43 which is greater than 20 authorized. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by fh-qube** [comment]
SonarQube analysis reported 12 issues:
* BLOCKER 5 blocker
* MAJOR 3 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/6aefc20d0bc953bbfcc1465d34040e035016faf6/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

File: lib/logger.js [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+  warn: false,
+  warning: false,
+  info: false,
+  debug: false
+};
+
+var doLog = function(level, prefix, msg) {
+  if( logCfg[level] ) {
+    // eslint-disable-next-line no-use-before-define
+    console.log(prefix, msg);
```
Comment (fh-qube) [inline lib/logger.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 7 issues:
* BLOCKER 1 blocker
* MAJOR 2 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/4d66f1f1b1c500387e4d1abde5d464ca6be7f61c/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 6 issues:
* BLOCKER 1 blocker
* MAJOR 1 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/179b40caa9ef041b3dbb46ef2ae7f99d9ea10584/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

**Comment by fh-qube** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 1 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [File "fhmongodb.js" has 550 lines, which is greater than 500 authorized. Split it into smaller files.](https://github.com/feedhenry/fh-db/blob/2bcc3b731428b640c4b6a0afffd044e8380622fe/lib/fhmongodb.js) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS104)

File: lib/utils.js [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+ *
+ * With the intention that it can be prepended with "mongodb://[username]:[password]@"
+ * and appended with "/[database][?options]
+ *
+ * @param host This is expected in the format: "host1" or "host1,host2" or ["host1", "host2"]
+ * @param port This is expected in the format: 123 or "123" or "123,456" or ["123", "456"] or [123, 456]
+ *
+ * @returns {string} host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+ */
+exports.parseConnectionUrl = function (host, port){
```
Comment (fh-qube) [inline lib/utils.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/utils.js [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+
+
+/**
+ * Returns the driver options URL encoded for the connection string; e.g.: w=1&j=true
+ *
+ * @param options an Array of options
+ *
+ * @returns {string}
+ */
+exports.parseConnectionUrlOptions = function (options) {
```
Comment (fh-qube) [inline lib/utils.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by fh-qube** [comment]
SonarQube analysis reported 10 issues:
* BLOCKER 2 blocker
* MINOR 6 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/eb249b9004d6614167d2c77e8b61ac9d504b714b/lib/fhmongodb.js#L174) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/eb249b9004d6614167d2c77e8b61ac9d504b714b/lib/fhmongodb.js#L166) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/eb249b9004d6614167d2c77e8b61ac9d504b714b/lib/fhmongodb.js#L186) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/eb249b9004d6614167d2c77e8b61ac9d504b714b/lib/fhmongodb.js#L99) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-qube** [comment]
SonarQube analysis reported 8 issues:
* MINOR 6 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/f927bccb86ddee514fcbf5d178d2f1270f2aa1fc/lib/fhmongodb.js#L99) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/f927bccb86ddee514fcbf5d178d2f1270f2aa1fc/lib/fhmongodb.js#L174) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/f927bccb86ddee514fcbf5d178d2f1270f2aa1fc/lib/fhmongodb.js#L166) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/f927bccb86ddee514fcbf5d178d2f1270f2aa1fc/lib/fhmongodb.js#L186) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

File: lib/mongo_compat_api.js [hunk 20]
```diff
@@ -0,0 +1,284 @@
...
+
+    return params.ditcher.database.close(force, callback);
+  };
+};
+
+/**
+ * The constructor function. Returns either an instance of the wrapper API
+ * around ditcher or a MongoDB handle as a promise.
+ *
+ * @param params required parameters
```
Comment (wei-lee) [inline lib/mongo_compat_api.js]: a description of what options are available in `params` here will be helpful

File: lib/utils.js [hunk 21]
```diff
@@ -0,0 +1,191 @@
+/**
+ * Parse a MongodB connection URL, split it up and return an object
+ * containing the configuration properties.
+ * @param mongoConnectionURL The connection URL as String
+ * @returns Configuration Object
+ */
+exports.parseMongoConnectionURL = function (mongoConnectionURL) {
```
Comment (wei-lee) [inline lib/utils.js]: not entirely sure about this function. This function is converting the connection string back to an object, only to allow the next function to convert the object back to connection string again.

Given that the database info is probably saved as object in fh-mbaas, perhaps `mongo_compat_api.js` can require the mongodb connection details to be passed in as an object?

**Comment by wei-lee** [comment]
LGTM. Some comments but excellent work. 👍

**Comment by fh-qube** [comment]
SonarQube analysis reported 8 issues:
* MINOR 6 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/53fff6db5f983722f5f3541680012c2a258d4180/lib/fhmongodb.js#L174) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/53fff6db5f983722f5f3541680012c2a258d4180/lib/fhmongodb.js#L166) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/53fff6db5f983722f5f3541680012c2a258d4180/lib/fhmongodb.js#L186) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/53fff6db5f983722f5f3541680012c2a258d4180/lib/fhmongodb.js#L99) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-qube** [comment]
SonarQube analysis reported 8 issues:
* MINOR 6 minor
* INFO 2 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/0feb5004d3f8433f58ecbcbf29d934a31db74a2e/lib/fhmongodb.js#L166) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/0feb5004d3f8433f58ecbcbf29d934a31db74a2e/lib/fhmongodb.js#L174) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/0feb5004d3f8433f58ecbcbf29d934a31db74a2e/lib/fhmongodb.js#L99) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-db/blob/0feb5004d3f8433f58ecbcbf29d934a31db74a2e/lib/fhmongodb.js#L186) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Pull request merged**